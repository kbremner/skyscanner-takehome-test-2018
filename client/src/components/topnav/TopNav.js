import React from 'react';
import BpkButton from 'bpk-component-button';
import BpkLargeMenuIcon from 'bpk-component-icon/lg/menu';
import PropTypes from '../../types';
import logo from './logo.svg';
import './TopNav.scss';

const TopNav = ({ onMenuClick }) => (
  <header className='topnav'>
    <a href="/">
      <span className='topnav__logo-text'>Skyscanner</span>
      <img className='topnav__icon-logo' alt="Skyscanner" src={logo} />
    </a>
    <BpkButton className="topnav__btn-menu" link onClick={onMenuClick}>
      <BpkLargeMenuIcon className="topnav__icon-menu" />
    </BpkButton>
  </header>
);

TopNav.propTypes = {
  onMenuClick: PropTypes.func.isRequired
};

export default TopNav;
