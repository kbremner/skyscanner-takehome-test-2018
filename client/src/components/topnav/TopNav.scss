@import "~bpk-mixins";

.topnav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-shrink: 0;
  padding: $bpk-spacing-sm $bpk-spacing-sm;
  background-color: $bpk-color-white;

  .topnav__logo-text {
    @include bpk-visually-hidden;
  }

  .topnav__icon-logo {
    vertical-align: middle;
  }

  .topnav__icon-menu {
    fill: $bpk-color-blue-500;
  }
}
