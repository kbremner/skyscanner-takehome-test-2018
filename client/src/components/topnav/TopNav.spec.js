import React from 'react';
import { shallow } from 'enzyme';
import TopNav from './TopNav';

const setup = () => {
  const onMenuClick = jest.fn();
  const wrapper = shallow(<TopNav onMenuClick={onMenuClick} />);
  return { wrapper, onMenuClick };
};

describe('TopNav', () => {
  it('renders', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls handler when menu button clicked', () => {
    const { wrapper, onMenuClick } = setup();
    wrapper.find('.topnav__btn-menu').simulate('click');
    expect(onMenuClick).toHaveBeenCalledTimes(1);
  });
});
