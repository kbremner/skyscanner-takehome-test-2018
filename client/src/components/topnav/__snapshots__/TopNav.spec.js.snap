// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav renders 1`] = `
<header
  className="topnav"
>
  <a
    href="/"
  >
    <span
      className="topnav__logo-text"
    >
      Skyscanner
    </span>
    <img
      alt="Skyscanner"
      className="topnav__icon-logo"
      src="test-file-stub"
    />
  </a>
  <BpkButton
    className="topnav__btn-menu"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={false}
    link={true}
    onClick={[MockFunction]}
    secondary={false}
    submit={false}
  >
    <Component
      className="topnav__icon-menu"
    />
  </BpkButton>
</header>
`;
